controllers/datadogagent/feature/oom_kill/feature.go

```diff
@@ -0,0 +1,98 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package oomkill
+
+import (
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/DataDog/datadog-operator/apis/datadoghq/v1alpha1"
+	"github.com/DataDog/datadog-operator/apis/datadoghq/v2alpha1"
+	apiutils "github.com/DataDog/datadog-operator/apis/utils"
+
+	apicommon "github.com/DataDog/datadog-operator/apis/datadoghq/common"
+	apicommonv1 "github.com/DataDog/datadog-operator/apis/datadoghq/common/v1"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/feature"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/object/volume"
+)
+
+func init() {
+	err := feature.Register(feature.OOMKillIDType, buildOOMKillFeature)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func buildOOMKillFeature(options *feature.Options) feature.Feature {
+	oomKillFeat := &oomKillFeature{}
+
+	return oomKillFeat
+}
+
+type oomKillFeature struct {
+	enable bool
+}
+
+// Configure is used to configure the feature from a v2alpha1.DatadogAgent instance.
+func (f *oomKillFeature) Configure(dda *v2alpha1.DatadogAgent) feature.RequiredComponents {
+	if dda.Spec.Features.OOMKill != nil && apiutils.BoolValue(dda.Spec.Features.OOMKill.Enabled) {
+		f.enable = true
+	}
+
+	return feature.RequiredComponents{
+		Agent: feature.RequiredComponent{IsRequired: &f.enable},
```

Comment:
here by setting the  `IsRequired: false` when `f.enable` is false make the `Agent` globally unwanted (with the current logic implemented in the Feature Factory). 

the idea was:
* set to `false`: we should not deploy this component
* nil: the feature don't need it, but don't care if it is deployed
* set to `true`: the feature need this component.

we can revisite the logic.

But I would suggest to no set it if `f.enable == false`

Reply:
right, thanks for catching that